java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowUtils.java

```diff
@@ -169,7 +171,11 @@ public static ArrowType getArrowTypeFromJdbcType(final JdbcFieldInfo fieldInfo,
       case Types.DECIMAL:
         int precision = fieldInfo.getPrecision();
         int scale = fieldInfo.getScale();
-        return new ArrowType.Decimal(precision, scale, 128);
+        if (precision > 38) {
```

Comment:
Is this correct? or should it be `precision + scale`? I'm not entirely sure how scale plays into how the values are stored.

Reply:
Should be right. Arrow Decimal128 maxes out at 38 as well.